### ISSUE TITLE
Ability to restore orders

### ISSUE THREAD

**Issue by Paul-Eraman-CoopCycle** [coopcycle/coopcycle-web#4333]
Currently we can restore tasks. This leads a user to believe that if they restore both a pickup and a dropoff of an order, the order is restored. However, it is not. 

Proposed UI: Add a right click option that is "restore order". 

if possible, also when restoring tasks, check if they are associated to non-cancelled order tasks. If this is the only unrestored task of the order, it should ask if you want to also restore the order.

**Comment by Paul-Eraman-CoopCycle** [comment]
could we just recycle the "duplicate order" feature that @vladimir-8  has created?

**Comment by vladimir-8** [comment]
Yes. So technically the old order will be still cancelled, we just create a new one with the same content. The only blocker for this to work might be https://github.com/coopcycle/coopcycle-web/issues/4613, if the time slot set previously is not valid NOW.

**Comment by Paul-Eraman-CoopCycle** [comment]
What would make it not valid? Having changed the time slot settings?

If so, could we just then take the user to the order creation page and any invalid setting is left blank and must be chosen again? Maybe with an informational message?

**Comment by vladimir-8** [comment]
What happens currently is that we pre-fill the first available time slot. If you "restore" an order with a time slot that's already started, it might not be displayed in the list of choices. That's exactly what happens for Hugo in https://github.com/coopcycle/coopcycle-web/issues/4613

**Comment by Paul-Eraman-CoopCycle** [comment]
would this not happen already, using the duplicate order feature as it is now?

**Comment by Paul-Eraman-CoopCycle** [comment]
got it. i think as long as it duplicates what it can and leaves the rest to be inputted in the new, duplicated order page, i think thats fine.
